### PR TITLE
Use the ggcr CheckPushPermission function in VerifyWriteAccess

### LINKED ISSUE #943: Prepare step will fail for https registries that use implicitly "local" urls
The `prepare` step will fail if using an https registry that has an implicitly local/insecure URL determined by [ggcr](https://github.com/google/go-containerregistry/blob/a0c4bd256482b8522065d5f6cf966281ef270680/pkg/name/registry.go#L65)

### PR DESCRIPTION
**Pull request by yaelharel**
Fixes #943

Signed-off-by: Yael Harel <yharel@vmware.com>

### CODE REVIEW

**Review by tomkennedy513 — APPROVED** [review]
So much cleaner!!

File: pkg/dockercreds/access_checker.go [hunk 1]
```diff
@@ -14,42 +13,12 @@ import (
...
-
-	if err = transport.CheckError(resp, http.StatusCreated, http.StatusAccepted); err != nil {
-		return err
-	}
-
-	return nil
+	return remote.CheckPushPermission(ref, keychain, http.DefaultTransport)
 }
 
 func VerifyReadAccess(keychain authn.Keychain, tag string) error {
```
Comment (tylerphelan) [inline pkg/dockercreds/access_checker.go:24]: I'm not sure if we need the `func diagnoseIfTransportError(err error) error` still. This appears to be handling an edge case for artifactory. I think we should remove this or add the artifactory test back. What's the context for this? @matthewmcnew

Comment (yaelharel) [reply pkg/dockercreds/access_checker.go:24]: I saw that we're using it in [VerifyReadAccess](https://github.com/pivotal/kpack/blob/44257a70c2c9e9ac5703eadc871c7c9be22cfadc/pkg/dockercreds/access_checker.go#L62).

Comment (yaelharel) [reply pkg/dockercreds/access_checker.go:24]: By the way, I don't know if it matters or not but [imgutil is using the ggcr Head function](https://github.com/buildpacks/imgutil/blob/56137f75e24d5b567db1dbb47ab05e958e257ec2/remote/remote.go#L476) in order to check read access while [kpack is using the ggcr Get function](https://github.com/pivotal/kpack/blob/44257a70c2c9e9ac5703eadc871c7c9be22cfadc/pkg/dockercreds/access_checker.go#L61).

Comment (tylerphelan) [reply pkg/dockercreds/access_checker.go:24]: `imgutil` is probably doing it correctly

**Review by matthewmcnew — APPROVED** [review]

File: pkg/dockercreds/access_checker.go (hunk 1 above)
Comment (tylerphelan) [reply pkg/dockercreds/access_checker.go:24]: using get is probably fine

Comment (tylerphelan) [reply pkg/dockercreds/access_checker.go:24]: let's just add `diagnoseIfTransportError` to the write check. using get should still be fine.

Comment (yaelharel) [reply pkg/dockercreds/access_checker.go]: Please see a67ce1afb9eda1ec9c5517379b6600b70381372b (I'll merge all of the commits before merging to `main`).

**Review by tylerphelan — APPROVED** [review]
